sherpa-ncnn/csrc/recognizer.cc

```diff
@@ -122,56 +123,55 @@ class Recognizer::Impl {
 #endif
 
   std::unique_ptr<Stream> CreateStream() const {
-    auto stream = std::make_unique<Stream>(config_.feat_config);
-    stream->SetResult(decoder_->GetEmptyResult());
-    stream->SetStates(model_->GetEncoderInitStates());
-    return stream;
-  }
-
-  std::unique_ptr<Stream> CreateStream(const char* contexts) const {
-    std::vector<std::vector<int32_t>> hotwords;
-    std::vector<int32_t> tmp;
-
-	  /*each line in hotwords file is a string which is segmented by space*/
-	  std::string hotwordsfile(contexts);
-    std::ifstream file1(hotwordsfile);
-    if (!file1) {
-      std::cerr << "open file failed: " << hotwordsfile << std::endl;
-      return nullptr;
-    }
-    std::string lines;
-	  std::string word;
-    while (std::getline(file1, lines)) {
-      std::istringstream iss(lines);
-	    while(iss >> word){
-	      std::cout<<word<<" ";
-        if (sym_.contains(word)) {
-          int number = sym_[word];
-          tmp.push_back(number);
-        } else {
-          std::cout << "can't find id" << std::endl;
-		      return nullptr;
+    if(config_.hotwordsfile.empty()) {
+      auto stream = std::make_unique<Stream>(config_.feat_config);
+      stream->SetResult(decoder_->GetEmptyResult());
+      stream->SetStates(model_->GetEncoderInitStates());
+      return stream;
+    } else {
+      std::vector<std::vector<int32_t>> hotwords;
+      std::vector<int32_t> tmp;
+
+	    /*each line in hotwords file is a string which is segmented by space*/
+      std::ifstream file1(config_.hotwordsfile);
```

Comment:
Could you read the file in the constructor of RecognizerImpl and save the content to a member variable `std::vector<int32_t> hotwords_`?

(Note: If the decoding method is greedy search, we don't need to read it even if the file exists).

If the file won't open, we can simply call `exit(-1)` after printing an error message.

Reply:
done